### DISCUSSION TITLE
Parcel packages missing in gatsby-parcel-config

### DISCUSSION THREAD

**Discussion by Primajin** [gatsbyjs/gatsby#37175] [Help] [answered]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

After upgrading to lockfile v3 I ran into errors when running `gatsby develop -S`:

```
 ERROR #11901  COMPILATION

Failed to compile Gatsby files (@parcel/core):

Could not determine version of @parcel/runtime-browser-hmr in
..\\..\\[redacted]\\node\_modules\\gatsby-parcel-config\\lib\\index.json. Either include it in
"dependencies" or "parcelDependencies"..

File path: C:\[redacted]\node_modules\gatsby-parcel-config\package.json

not finished compile gatsby files - 1.678s
```

when adding it into `node_modules\gatsby-parcel-config\package.json` the next ones show up:

```
Could not determine version of @parcel/runtime-service-worker in [...]
Could not determine version of @parcel/runtime-react-refresh in [...]
```

When installing them myself as dev dependencies the compilation works fine again!
So I think they seem to be missing in [packages/gatsby-parcel-config/package.json](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-parcel-config/package.json)

### Reproduction Link

https://codesandbox.io/s/misty-star-2w8xc2

### Steps to Reproduce

1. Given you have npm > 7 and a lockfile version 3
2. Install dependencies
3. run gatsby develop

(this could not be reproduced in CodeSandbox or Stackblitz since they are on old npm 6)

### Expected Result

Gatsby compiles and runs

### Actual Result

Gatsby fails to compile since peer dependencies are missing

### Workaround

Adding these to dev dependencies.

```json
  "devDependencies": {
    "@parcel/runtime-browser-hmr": "2.6.2",
    "@parcel/runtime-react-refresh": "2.6.2",
    "@parcel/runtime-service-worker": "2.6.2",
```

### Environment

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (12) x64 AMD Ryzen 5 5600G with Radeon Graphics         
  Binaries:
    Node: 19.2.0 - C:\Program Files\nodejs\node.EXE
    npm: 9.1.3 - C:\Program Files\nodejs\npm.CMD
  Languages:
    Python: 3.10.5 - /c/Python310/python
  Browsers:
    Edge: Spartan (44.22621.900.0), Chromium (107.0.1418.62)
  npmPackages:
    gatsby: ^4.24.8 => 4.24.8 
    gatsby-plugin-csp: ^1.1.4 => 1.1.4 
    gatsby-plugin-emotion: ^7.24.0 => 7.24.0 
    gatsby-plugin-gatsby-cloud: ^4.24.1 => 4.24.1 
    gatsby-plugin-image: ^2.24.0 => 2.24.0 
    gatsby-plugin-manifest: ^4.24.0 => 4.24.0 
    gatsby-plugin-offline: ^5.24.0 => 5.24.0 
    gatsby-plugin-react-helmet: ^5.24.0 => 5.24.0 
    gatsby-plugin-robots-txt: ~1.7.1 => 1.7.1 
    gatsby-plugin-sitemap: ^5.24.0 => 5.24.0 
    gatsby-plugin-web-font-loader: ^1.0.4 => 1.0.4 
    gatsby-source-prismic: ^5.2.10 => 5.2.10
```

**Comment by Primajin** [comment]
Added:

### Workaround

Adding these to dev dependencies.

```json
  "devDependencies": {
    "@parcel/runtime-browser-hmr": "2.6.2",
    "@parcel/runtime-react-refresh": "2.6.2",
    "@parcel/runtime-service-worker": "2.6.2",
```

**Comment by LekoArts** [comment]
Hi!

The version you should require (that is also in your lockfile) doesn't have these in their config: https://unpkg.com/browse/gatsby-parcel-config@0.15.1/lib/index.json

So they also shouldn't be required when Parcel runs. Can you make sure that your actual `node_modules/gatsby-parcel-config` looks correct?

**Reply by Primajin** [reply]
Hmm yes my local node_modules/gatsby-parcel-config looks exactly like the one you linked

**Comment by Primajin** [comment, marked as answer]
To my shame I must admit the solution was so easy that I completely overlooked it - `gatsby clean` did the trick... 🤦🏻‍♂️
